MTA-6893 Document Hub OIDC authentication (#382)
migtools/mta-documentation: docs/topics/mta-install/proc_enable-authentication-cr.adoc

:_newdoc-version: 2.18.5
:_template-generated: 2025-07-31
:_mod-docs-content-type: PROCEDURE

[id="enable-authentication-cr_{context}"]
= Enable authentication in Tackle custom resource

[role="_abstract"]

> Administrators can optionally configure authentication fields and token lifespan in the Tackle custom resource. If you do not configure, the {ProductFullName} Hub applies default values for the authentication and token fields.

Reviewer: ```suggestion
As an administrator, you can configure authentication fields and the token lifespan in the Tackle custom resource (CR). If you do not configure these fields, the {ProductFullName} Hub applies default values.
```

The word "optionally" is redundant because the auxiliary verb "can" combined with the next sentence ("If you do not...") already implies it is optional. Removing it improves minimalism. Additionally, you can tighten the end of the second sentence to just "applies default values" to avoid needlessly repeating the field names.

  hub_oidc_refresh_token_lifespan:172800
  hub_apikey_lifespan:87600
  hub_auth_cache_lifespan:5
  hub_ldap_auth_lifespan:5
  hub_oidc_key_rotation:90
----
+
* `feature_auth_required`:: Enforces authentication to the Hub API and the web console. By default, the value is `true`.
* `idp_primary`:: If you set this field `true`, {ProductShortName} Hub automatically redirects authentication to the auto-discovered Keycloak Identity Provider in upgraded deployments where users authenticated using Keycloak. If the field value is `false`, users can authenticate on the {ProductShortName} login page. If you configured an identity provider (IdP), then users can authenticate by using the IdP button on the login page.
* `hub_oidc_token_lifespan`:: By default, the OAuth access token lifespan is 300 seconds.
* `hub_oidc_refresh_token_lifespan`:: By default, the OAuth refresh token lifespan is 172800 seconds or 2 days.
* `hub_apikey_lifespan`:: By default, the OAuth API key lifespan is 87600 hours.
* `hub_auth_cache_lifespan`:: By default, the OAuth authentication cache lifespan is 5 minutes.
* `hub_ldap_auth_lifespan`:: By default, the LDAP authentication lifespan is 5 minutes.

> * `hub_oidc_key_rotation`:: By default, the OIDC key rotation interval is 90 days.

Reviewer: ```suggestion
* `feature_auth_required`:: Specifies whether to enforce authentication to the Hub API and the web console. The default value is `true`.
* `idp_primary`:: Specifies whether the {ProductShortName} Hub automatically redirects authentication to the auto-discovered Keycloak identity provider in upgraded deployments where users authenticated by using Keycloak. When this value is `true`, the Hub redirects users automatically. When `false`, users can authenticate on the {ProductShortName} login page. If you configured an identity provider (IdP), users can authenticate by using the IdP button on the login page.
* `hub_oidc_token_lifespan`:: Specifies the OAuth access token lifespan. The default value is 300 seconds.
* `hub_oidc_refresh_token_lifespan`:: Specifies the OAuth refresh token lifespan. The default value is 172800 seconds (2 days).
* `hub_apikey_lifespan`:: Specifies the OAuth API key lifespan. The default value is 87600 hours.
* `hub_auth_cache_lifespan`:: Specifies the OAuth authentication cache lifespan. The default value is 5 minutes.
* `hub_ldap_auth_lifespan`:: Specifies the LDAP authentication lifespan. The default value is 5 minutes.
* `hub_oidc_key_rotation`:: Specifies the OIDC key rotation interval. The default value is 90 days.
```
